### ISSUE TITLE
Explain binary categorization results with a LogisticRegression model

### ISSUE THREAD

**Issue by rth** [FreeDiscovery/FreeDiscovery#33]
Cf. issue #32 for general context.

This issue aims to add the ability to explain categorization results for Bag of Words (BOW) non hashed feature extraction (requires #11 fixed) using the Logistic Regression¹ model. This would probably be the simplest case, that could be later generalized.

The decision function for the logistic regression (cf. [`sklearn.linear_model.LogisticRegression`](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html) is computed as 
```
    df =  intercept_ + Σ coef_[i] * X[i]
```
where `coef_` is a 1D array of LogisticRegression model coefficients, and `X` the 1D feature vector (e.g. tf-idf weights). The probability for a document to be e.g. relevant is then given by `sigmoid_func(df)`. 

Therefore the weight of each word (or n-gram) in the final classification result can be estimated with `coef_[i] * X[i]`. 

This issue would thus involve adding a `explain(self, doc_id)` method to the [`Categorizer` class](https://github.com/FreeDiscovery/FreeDiscovery/blob/2d1f10ea821f6f269c7fce49c6ecbf3854a1bfa6/freediscovery/categorization.py#L25) that given a trained ML model, and a document id (`doc_id`) returns a dictionary 
```
{'word_1_in_the_document': <coef_[i]*X[i]>_for_that_word, 'word_2_in_the_document': ...}
```

Given a LogisticRegression model trained on BOW, non hashed features, the general workflow (at later stages) could then be to,

 - [x] Calculate  the feature importance with `Categorizer.explain`
 - [x] Find a way to visualize the feature importance, e.g. by printing the text with as background color for each word it's importance in the model. 
     * this might require to rerun the FeatureExtraction or a custom tokenizer again on the given document, then find a correspondence between the features id and the words/n-grams in that document. 

**Note:** it might be interesting to use L1 penalty for the LogisticRegression as it tend to result in sparser (and thus easier to visualize) coefficients [[1](http://scikit-learn.org/stable/auto_examples/linear_model/plot_logistic_l1_l2_sparsity.html)], [[2](https://www.quora.com/Why-is-L1-regularization-supposed-to-lead-to-sparsity-than-L2)].

¹ coefficients in the SVM model are more difficult to interpret

**Comment by DenisFLASH** [comment]
@rth so this issue requires #11 to be fixed, which in its turn requires #26?   if it is true, i'll take 26 (which i wanted to take 2 weeks ago before we noticed that Metrics API can't be closed due to the absence of the docs).

Or, if this issue is more critical and needs some progress even with 11 and 26 not fixed, i can start this one.  What do you think?   
P.S. Merry Christmas ;)

**Comment by rth** [comment]
Fixed in PR #80.
